task6/script.js

```diff
@@ -0,0 +1,334 @@
+        },
+        {
+            id: '20',
+            description: 'super desc 20',
+            createdAt: new Date('2020-03-17T12:12:12'),
+            author: 'Agent 20',
+            likes: ['Peter', 'Arnold', 'Stalin'],
+            hashTags: ['USSR', 'Stalin']
+        },
+    ];
+
+    const postProperties = [
+        'id',
+        'description',
+        'createdAt',
+        'author',
+        'photoLink',
+        'likes',
+        'hashTags'
+    ];
+
+    function validatePost(post) {
+        let optionalFieldAmount = 0;
+        if (!post.hasOwnProperty('id') || typeof post.id !== 'string') {
+            return false;
+        }
+
+        if (!post.hasOwnProperty('description') || typeof post.description !== 'string'
+            || post.description.length >= 200) {
+            return false;
+        }
+
+        if (!post.hasOwnProperty('createdAt') || !(post.createdAt instanceof Date)) {
+            return false;
+        }
+
+        if (!post.hasOwnProperty('author') || typeof post.author !== 'string') {
+            return false;
+        }
+
+        if (!post.hasOwnProperty('photoLink') || typeof post.photoLink !== 'string') {
+            optionalFieldAmount--;
+        }
+
+        if (!post.hasOwnProperty('likes') || !Array.isArray(post.likes)) {
+            return false;
+        }
+        else {
+            if (!post.likes.every(function(element, index, array) {
+                return typeof element === 'string';
+            })) {
+                optionalFieldAmount++;
+            }
+        }
+
+        if (!post.hasOwnProperty('hastTags') || !Array.isArray(post.hashTags)) {
+            optionalFieldAmount--;
+        }
+        else {
+            if (!post.hastTags.every(function(element, index, array) {
+                return typeof element === 'string';
+            })) {
+                optionalFieldAmount++;
+             }
+        }
+
+        return post.keys().length === postProperties.length - optionalFieldAmount;
+    }
+
+    function compareDates(a, b) {
+        return b.createdAt - a.createdAt;
+    }
+
+
+    function validateFilerConfig(filterConfig) {
+
+    }
+
+    function isValid(post, filterConfig) {
+        if (filterConfig.author !== '' && filterConfig.author !== post.author) {
+            return false;
+        }
+
+        return !(filterConfig.createdAt instanceof Date && compareDates(post.createdAt, filterConfig.createdAt) > 0);
+    }
+
+    function getPosts(skip = 0, top = 10, filterConfig) {
+        posts.sort(compareDates);
+        let newList = [];
+        validateFilerConfig(filterConfig);
+        for (let i = skip; i < posts.length && i < skip + top; ++i) {
+            if (isValid(posts[i], filterConfig)) {
+                newList.push(posts[i]);
+            }
+        }
+        return newList;
+    }
+
+    function getPost(id) {
+        if (id && typeof id === 'string') {
+            for (let postObject in posts) {
```

Comment:
`Find` method can be used here https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/find